vm/src/obj/objtype.rs

```diff
@@ -72,43 +72,70 @@ impl PyClassRef {
             new_name
         })
     }
-}
 
-#[pyimpl(flags(BASETYPE))]
-impl PyClassRef {
     pub fn iter_mro(&self) -> impl Iterator<Item = &PyClassRef> + DoubleEndedIterator {
         std::iter::once(self).chain(self.mro.iter())
     }
 
+    pub(crate) fn first_in_mro<F, R>(&self, f: F) -> Option<R>
+    where
+        F: Fn(&Self) -> Option<R>,
+    {
+        f(self).or_else(|| self.mro.iter().filter_map(|cls| f(cls)).next())
```

Comment:
```suggestion
        self.iter_mro().find_map(|cls| f(cls))
```

Reply:
Due to the different expectation of the cost, I changed its form and added comments